usb: udc_virtual: Adapt to new API

finikorg opened the pull request:
- Fixes build when udc_virtual was using old API.
- Fixes build for 64 bit platforms

finikorg committed:
usb: udc_virtual: Adapt to updated API

Commit 00adb2a539a8 ("drivers: udc: remove no more required pending
state flag") removed state parameter from udc_buf_peek and udc_buf_get.
Adapt udc virtual driver so the code compiles.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

tmon-nordic reviewed (commented):
Generally good, but the commit message is not clear. I would prefer something like:

```
usb: udc_virtual: Adapt to updated API

Commit 00adb2a539a8 ("drivers: udc: remove no more required pending state flag") removed state parameter from udc_buf_peek and udc_buf_get. Adapt udc virtual driver so it compiles.
```

finikorg commented:
> Generally good, but the commit message is not clear. I would prefer something like:
> 
> ```
> usb: udc_virtual: Adapt to updated API
> 
> Commit 00adb2a539a8 ("drivers: udc: remove no more required pending state flag") removed state parameter from udc_buf_peek and udc_buf_get. Adapt udc virtual driver so it compiles.
> ```

Fixed this way.

finikorg committed:
samples: usb: Allow test to build in CI

Remove dependency on usb_device for virtual USB device / hosts sample.
This allows sample to be built in CI.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

finikorg committed:
usb: Fix build for 64 bit platforms

Use proper format for size_t type eliminating warnings of type:

...
error: format '%u' expects argument of type 'unsigned int', but
argument 2 has type 'size_t' {aka 'long unsigned int'}
[-Werror=format=]
...

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

finikorg commented on samples/subsys/usb/shell/sample.yaml:
```diff
@@ -18,5 +18,5 @@ tests:
     extra_args: CONF_FILE="device_and_host_prj.conf"
                 OVERLAY_CONFIG="virtual.conf"
                 DTC_OVERLAY_FILE="virtual.overlay"
-    platform_allow: qemu_cortex_m3
```
OK, I will fix this way, for now I will remove commits in question and make separate PR. There is benefit for adding more qemu type, at least for qemu_x86_64, as it found some print format issues.

finikorg replied:
@jfischer-no I have removed this commit for now from PR

finikorg replied:
> This actually not right (my fault), there are board specific overlays, correct fix would be to remove OVERLAY_CONFIG and DTC_OVERLAY_FILE args first, then add board specific overlays for qemu_x86 qemu_x86_64 (btw, I do not see any benefit to add additional qemu type).

Removed as you have recommended since the test was failing otherwise.

finikorg commented on samples/subsys/usb/shell/virtual.overlay:
```diff
@@ -4,8 +4,6 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-/delete-node/ &zephyr_udc0;
```
Yes, but using virtual USB devices makes sense only for qemu or native_posix, right?

jfischer-no replied:
No, it works well on real hardware, and there are nice use cases for it.

finikorg replied:
@jfischer-no I removed this commit from PR

finikorg committed:
samples: usb: Remove unneeded old configuration

Remove OVERLAY_CONFIG and DTC_OVERLAY_FILE from sample configuration.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>
